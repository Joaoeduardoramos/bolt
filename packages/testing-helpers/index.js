export { isConnected } from './is-connected';
import {
  render as renderServer,
  renderString as renderStringServer,
  stop as stopServer,
} from '@bolt/twig-renderer';
export { fixture as html } from '@open-wc/testing-helpers';

export async function render(template, data) {
  return await renderServer(template, data, true);
}

export async function renderString(template, data) {
  return await renderStringServer(template, data, false);
}

export { stopServer };

// import {
//   // isConnected,
//   // render,
//   // renderString,
//   // stopServer,
//   html,
// } from '../../../testing-helpers';
