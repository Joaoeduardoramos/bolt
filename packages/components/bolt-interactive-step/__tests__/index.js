import {
  render,
  renderString,
  stop as stopTwigRenderer,
} from '@bolt/twig-renderer';
import { fixture as html } from '@open-wc/testing-helpers';

const { readYamlFileSync } = require('@bolt/build-tools/utils/yaml');
const { join } = require('path');
const schema = readYamlFileSync(
  join(__dirname, '../interactive-step.schema.yml'),
);
const { tag } = schema.properties;

async function renderTwig(template, data) {
  return await render(template, data, true);
}

async function renderTwigString(template, data) {
  return await renderString(template, data, true);
}

const timeout = 60000;

describe('interactive step', async () => {
  let page;

  afterAll(async () => {
    await stopTwigRenderer();
  }, timeout);

  beforeEach(async () => {
    page = await global.__BROWSER__.newPage();
    await page.goto('http://127.0.0.1:4444/', {
      timeout: 0,
      waitLoad: true,
      waitNetworkIdle: true, // defaults to false
    });
  }, timeout);

  test('basic interactive-step component renders', async () => {
    const results = await renderTwig(
      '@bolt-components-interactive-step/interactive-step.twig',
    );
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  test('basic interactive-step component with the global `no-shadow` prop added', async () => {
    const results = await renderTwig(
      '@bolt-components-interactive-step/interactive-step.twig',
      {
        no_shadow: true,
      },
    );
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  test('interactive-step with outer CSS class via Drupal Attributes', async () => {
    const results = await renderTwig(
      '@bolt-components-interactive-step/interactive-step.twig',
      {
        attributes: {
          class: ['u-bolt-margin-top-medium'],
        },
      },
    );
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  test('Default <bolt-interactive-step> w/o Shadow DOM renders', async function() {
    const renderedComponentHTML = await page.evaluate(() => {
      const component = document.createElement('bolt-interactive-step');
      component.textContent = 'interactive-step Test';
      document.body.appendChild(component);
      component.useShadow = false;
      component.updated();
      return component.outerHTML;
    });

    const renderedHTML = await html(renderedComponentHTML);

    const image = await page.screenshot();
    expect(image).toMatchImageSnapshot({
      failureThreshold: '0.01',
      failureThresholdType: 'percent',
    });

    expect(renderedHTML).toMatchSnapshot();
  });

  test('Default <bolt-interactive-step> with Shadow DOM renders', async function() {
    const defaultComponentShadowRoot = await page.evaluate(() => {
      const component = document.createElement('bolt-interactive-step');
      component.textContent =
        'interactive-step Component Test -- Shadow Root HTML';
      document.body.appendChild(component);
      component.updated();
      return component.renderRoot.innerHTML;
    });

    const defaultComponentOuter = await page.evaluate(() => {
      const component = document.createElement('bolt-interactive-step');
      component.textContent = 'interactive-step Component Test -- Outer HTML';
      document.body.appendChild(component);
      component.updated();
      return component.outerHTML;
    });

    const renderedShadowDomHTML = await html(defaultComponentShadowRoot);
    const renderedHTML = await html(defaultComponentOuter);

    expect(renderedHTML.textContent).toEqual(
      'interactive-step Component Test -- Outer HTML',
    );

    const image = await page.screenshot();

    expect(image).toMatchImageSnapshot({
      failureThreshold: '0.01',
      failureThresholdType: 'percent',
    });

    expect(renderedShadowDomHTML).toMatchSnapshot();
    expect(renderedHTML).toMatchSnapshot();
  });
});
