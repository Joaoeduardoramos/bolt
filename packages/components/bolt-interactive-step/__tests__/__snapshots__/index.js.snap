// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`interactive step Default <bolt-interactive-step> w/o Shadow DOM renders 2`] = `
<bolt-interactive-step
  style=""
>
  <!---->
  
      
      
  <li
    class="c-bolt-interactive-step"
    data-step="1"
    is="shadow-root"
    style=""
  >
    
        
    <div
      class="c-bolt-interactive-step__nav-item-wrapper"
      style=""
    >
      
          
      <span
        class="c-bolt-interactive-step__dot"
        style=""
      >
        ‚óè
      </span>
      
          
      <span
        class="c-bolt-interactive-step__title"
        style=""
      >
        <!---->
        <!---->
        <!---->
        <!---->
      </span>
      
        
    </div>
    
        
    <div
      class="c-bolt-interactive-step__body"
      data-active="false"
      style=""
    >
      
          
      <div
        class="c-bolt-interactive-step__dialogue c-bolt-interactive-step__dialogue--left"
        style=""
      />
      
          
      <div
        class="c-bolt-interactive-step__interaction-images"
        style=""
      >
        
            
        <span
          class="c-bolt-interactive-step__interaction-band"
          style=""
        >
          <!---->
          <!---->
          <!---->
          <!---->
        </span>
        
            
        <img
          alt="Customer is happy"
          class="c-bolt-interactive-step__avatar"
          src="https://github.com/basaltinc/temp-pega-dummy-assets/raw/master/customer-happy.png"
          style=""
        >
          
            
        </img>
        <img
          alt="Your helpful Pega Rep"
          class="c-bolt-interactive-step__avatar c-bolt-interactive-step__avatar--right"
          src="https://github.com/basaltinc/temp-pega-dummy-assets/raw/master/pega-rep.png"
          style=""
        >
          
          
        </img>
      </div>
      
          
      <div
        class="c-bolt-interactive-step__step-body"
        style=""
      >
        
            
        <!---->
        <!---->
        
          
      </div>
      
        
    </div>
    
      
  </li>
  
    
  <!---->
</bolt-interactive-step>
`;

exports[`interactive step Default <bolt-interactive-step> with Shadow DOM renders 2`] = `
<style
  style=""
>
  
            bolt-interactive-step{display:block}bolt-interactive-step:not(:last-child){margin-bottom:1.65rem}.c-bolt-interactive-step{position:relative;padding-left:40px;text-align:left;cursor:pointer;font-weight:bolder;-webkit-transition:font-size .5s;transition:font-size .5s}@media (min-width:37.5em){.c-bolt-interactive-step{position:static}.c-bolt-interactive-step__body{position:absolute;top:0;right:0;width:calc(100% - 300px)}}.c-bolt-interactive-step__nav-item-wrapper{display:-webkit-box;display:flex;-webkit-box-align:center;align-items:center}.c-bolt-interactive-step__dot{display:inline-block;position:absolute;z-index:10;margin-left:-38px;color:#aab2f2;cursor:pointer;font-size:20px;-webkit-transition:color .5s,-webkit-transform .5s;transition:color .5s,-webkit-transform .5s;transition:color .5s,transform .5s;transition:color .5s,transform .5s,-webkit-transform .5s}.c-bolt-interactive-step__title{display:inline-block;cursor:pointer;-webkit-transition:color .5s,-webkit-transform .5s;transition:color .5s,-webkit-transform .5s;transition:color .5s,transform .5s;transition:color .5s,transform .5s,-webkit-transform .5s}.c-bolt-interactive-step__body{display:-webkit-box;display:flex;-webkit-box-orient:vertical;-webkit-box-direction:normal;flex-direction:column;-webkit-box-align:center;align-items:center;height:0;overflow:hidden;-webkit-transition:height 2s ease-in-out;transition:height 2s ease-in-out}.c-bolt-interactive-step__body[data-active=true]{height:auto}.c-bolt-interactive-step__interaction-images{display:-webkit-box;display:flex;-webkit-box-pack:center;justify-content:center;-webkit-box-align:center;align-items:center;position:relative;max-width:250px}.c-bolt-interactive-step__interaction-band{display:-webkit-box;display:flex;-webkit-box-pack:center;justify-content:center;-webkit-box-align:center;align-items:center;height:100px;padding:20px}.c-bolt-interactive-step__avatar{position:absolute;left:0;width:57px;height:57px}.c-bolt-interactive-step__avatar--right{right:0;left:unset}.c-bolt-interactive-step__dialogue{position:relative;width:100%;max-width:250px;height:50px}.c-bolt-interactive-step--active .c-bolt-interactive-step__dot{-webkit-transform:scale(1.5);transform:scale(1.5);color:orange}.c-bolt-interactive-step--active .c-bolt-interactive-step__title{-webkit-transform:scale(1.1) translateX(4px);transform:scale(1.1) translateX(4px);color:orange}
          
</style>
`;

exports[`interactive step Default <bolt-interactive-step> with Shadow DOM renders 3`] = `
<bolt-interactive-step
  style=""
>
  interactive-step Component Test -- Outer HTML
</bolt-interactive-step>
`;

exports[`interactive step basic interactive-step component renders 1`] = `
<bolt-interactive-step>
  <div class="c-bolt-interactive-step"
       is="shadow-root"
  >
  </div>
</bolt-interactive-step>
`;

exports[`interactive step basic interactive-step component with the global \`no-shadow\` prop added 1`] = `
<bolt-interactive-step no-shadow>
  <div class="c-bolt-interactive-step"
       is="shadow-root"
  >
  </div>
</bolt-interactive-step>
`;

exports[`interactive step interactive-step with outer CSS class via Drupal Attributes 1`] = `
<bolt-interactive-step class="u-bolt-margin-top-medium">
  <div class="c-bolt-interactive-step"
       is="shadow-root"
  >
  </div>
</bolt-interactive-step>
`;
